lib/tasks/update_participant_profile_states.rake

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require "rake"
+
+namespace :update_participant_profile_states do
+  desc "Back filling cpd_lead_providers on participant_profile_states"
+  task populate_cpd_lead_provider: :environment do
+    PaperTrail.request.controller_info = {
+      reason: "Back filling cpd_lead_providers on participant_profile_states",
+    }
+
+    participant_profiles_with_a_single_induction_record = ParticipantProfile.joins(:induction_records).group(:id).having("count(*) < 2")
+
+    participant_profiles_with_a_single_induction_record.each do |pp|
+      pp.participant_profile_states.each do |ps|
+        ps.update_column(:cpd_lead_provider_id, pp.school&.partnerships&.first&.lead_provider&.cpd_lead_provider&.id)
```

Comment:
I had quick exchange with @tonyheadford since you already have the induction record you should be able to figure out the provider from the partnership. so something like `participant_profile.induction_records.active.latest.induction_programme.partnership...`

Reply:
👍